pkg/controller/accountclaim/reuse.go

```diff
@@ -44,6 +44,22 @@ func (r *ReconcileAccountClaim) finalizeAccountClaim(reqLogger logr.Logger, acco
 		return err
 	}
 
+	// Get latest version of the account
+	refreshedAccount, err := r.getClaimedAccount(accountClaim.Spec.AccountLink, awsv1alpha1.AccountCrNamespace)
```

Comment:
Do we get an error if we don't refresh the account given the above two functions are updating the account CR? Also I'm not sure we need a new var here for the Account CR.